### PR TITLE
Upgrade Babel@6

### PR DESCRIPTION
**Pull request by rcorral**
Changed the way Util.mixin works. Because we use webpack, it is not aware of the prototypal inheritance that we dynamically try to build. So therefore using `super` doesn't work as expected.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @mlunoe

File: src/Confirm/Confirm.js [hunk 1]
```diff
@@ -95,3 +95,5 @@ Confirm.propTypes = {
   rightButtonClassName: PropTypes.string,
   rightButtonCallback: PropTypes.func.isRequired
 };
+
+module.exports = Confirm;
```
Comment (rcorral) [inline src/Confirm/Confirm.js]: Have to do this now when exporting. Apparently Babel 5 was doing it wrong and they've changed the way that `export default` works, so this is why there's many files that have this change.

File: src/Util/DOMUtil.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-let computeInnerBound = (compstyle, acc, key) => {
+var computeInnerBound = function (compstyle, acc, key) {
```
Comment (rcorral) [inline src/Util/DOMUtil.js]: Made this change because tests just weren't passing for me. ¯\_(ツ)_/¯

File: src/Util/__tests__/Util-test.js [hunk 3]
```diff
@@ -68,7 +68,7 @@ describe('Util', function () {
     });
 
     it('doesn\'t exclude any properties', function () {
-      let expectedResult = {
+      var expectedResult = {
```
Comment (rcorral) [inline src/Util/__tests__/Util-test.js]: Same thing here, tests weren't passing because of the `let`s.
¯\_(ツ)_/¯

File: .webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+module.exports = {
+  devtool: 'source-map',
+  entry: glob.sync(config.files.docs.srcJS),
+  output: {filename: config.files.docs.distJS},
+  module: {
+    loaders: [{
+      exclude: /node_modules/,
+      loader: 'babel-loader',
+      query: {
+        cacheDirectory: true,
```
Comment (kennyt) [inline .webpack.config.js]: nice! the previous gulpfile didn't have any caching at all.

File: docs/gulpTasks.js [hunk 5]
```diff
@@ -112,14 +115,14 @@ gulp.task('docs:webpack', ['docs:eslint'], function (callback) {
       timing: true
     }));
 
-    browserSync.reload();
```
Comment (kennyt) [inline docs/gulpTasks.js]: why do we no longer do `reload` in webpack update?

**Comment by kennyt** [comment]
:+1:

File: docs/gulpTasks.js [hunk 6]
```diff
@@ -29,15 +33,18 @@ gulp.task('docs:browsersync', function () {
   });
 });
 
-gulp.task('docs:eslint', ['eslint'], function () {
-  return gulp.src([config.dirs.docs.srcJS + '/**/*.?(js|jsx)'])
+// Create a function so we can use it inside of webpack's watch function.
+function eslintFn () {
```
Comment (mlunoe) [inline docs/gulpTasks.js]: #nit There is an extra space here

File: docs/gulpTasks.js [hunk 7]
```diff
@@ -69,11 +80,15 @@ gulp.task('docs:minify-js', ['docs:replace-js-strings'], function () {
     .pipe(gulp.dest(config.dirs.docs.distJS));
 });
 
-gulp.task('docs:replace-js-strings', ['docs:webpack'], function () {
+function replaceJsStringsFn () {
```
Comment (mlunoe) [inline docs/gulpTasks.js]: #nit There is an extra space here

**Pull request merged**